src/NuGet.Core/NuGet.Resolver/PackageResolver.cs

```diff
@@ -42,6 +43,30 @@ public IEnumerable<PackageIdentity> Resolve(PackageResolverContext context, Canc
                 }
             }
 
+            // validate existing package.config for any invalid dependency
+            var invalidsPackagesLst = new List<string>();
+            var installedPackages = context.PackagesConfig.Select(p => p.PackageIdentity);
+            foreach (var package in installedPackages)
+            {
+                var existingPackage = context.AvailablePackages.FirstOrDefault(p => StringComparer.OrdinalIgnoreCase.Equals(p.Id, package.Id) && p.Version.Equals(package.Version));
+                if(existingPackage != null)
+                {
+                    // check if each dependency can be satisfied with existing packages
+                    var brokenDependencies = GetBrokenDependencies(existingPackage, installedPackages);
+                    if(brokenDependencies !=  null && brokenDependencies.Any())
+                    {
+                        invalidsPackagesLst.AddRange(brokenDependencies.Select(dependency => FormatDependencyConstraint(existingPackage, dependency)));
```

Comment:
suggestion: can we split this into two lines to increase readability? I am personally not a big fan of chained linq queries, they make understanding code a little harder.